Use <vulkan/vulkan.hpp> as precompiled header.

asuessenbach opened the pull request:
## Description

Use <vulkan/vulkan.hpp> as precompiled header with all the hpp-based samples and the framework, expecting build times to be reduced.

Tested on Windows 11.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

asuessenbach committed:
Use <vulkan/vulkan.hpp> as precompiled header.

asuessenbach committed:
Use <vulkan/vulkan.hpp> as precompiled header.

SaschaWillems commented:
Just gave this a spin with VS2019 on Windows 10, and it doesn't compile for me. I get the same errors as in the CI build:

```fatal error C1010: unexpected end of file while looking for precompiled header. Did you forget to add '#include "pch.h"' to your source?```

gpx1000 commented:
It does indeed work in Linux.  I'm wondering if there's a bug in the CMake version being used.  target_precompile_headers wasn't introduced until CMake 3.16 and our base minimum version is 3.12.  The windows build not using /fi to avoid needing to include the precompiled header is probably to do with not having a > CMake 3.16 designation maybe?  Although it might be a bug elsewhere in CMake as 3.24 looks like the version that CICD uses.

gpx1000 committed:
Auto enable vkb enable portability (#497)

* Fix issue #452

* Attempt some logic to make setting of VKB_ENABLE_PORTABILITY happen if and only if Vulkan_VERSION is at least 1.3 which should include the required portability extension which will enable the instance to find the non-conformant vulkan for moltenVK.

* update copyright.

* make the compile definition explicit.

* Validated as working with Vulkan 1.3.216.0 on MacBook Air M2.

* enable vulkan.hpp samples to work.

* don't crash open_gl_interop.cpp if a gl_context doesn't get created.  As in the case when extensions required are not present.

* revert the 3.19 cmake version upgrade for platforms other than OSX.
NB: This *might* mean that Android on OSX would have an issue due to the version of CMake that Android SDK might ship with is 3.18.  Revisit this once most people are on Vulkan 1.3 and/or confirmation that android on OSX is a problem.

* revert the 3.19 cmake version upgrade for platforms other than OSX.
NB: This *might* mean that Android on OSX would have an issue due to the version of CMake that Android SDK might ship with is 3.18.  Revisit this once most people are on Vulkan 1.3 and/or confirmation that android on OSX is a problem.

* update the Copyright

* update with clang-format

* oddly running clang-format locally resulted in some white space changes which don't agree with the CICD white space results.

asuessenbach commented:
That's strange, I don't have that issue on Windows, using CMake 3.23 or CMake 3.24.
For testing, I'll remove the call to vulkan_samples_pch in framework/CMakeLists.txt, to see if there's some interference.

asuessenbach committed:
Test: remove vulkan_samples_pch from framework/CMakeLists.txt

asuessenbach committed:
Test: remove vulkan_samples_pch from framework/CMakeLists.txt

asuessenbach committed:
Add transcoding of api sample hlsl_shaders based on vulkan.hpp (#512)

* Change vkb::core::HPPPhysicalDevice and vkb::core::HPPInstance from facades over vkb::PhysicalDevice and vkb::Instance, respectively, to self-contained classes using vulkan.hpp

* Explicitly assign HPPStructurType::structureType to a variable to be usable in std::map::find().

* Change vkb::core::HPPVulkanResource from a facade over vkb::core::VulkanResource to a self-contained class using vulkan.hpp

* Add transcoding of api sample terrain_tessellation based on vulkan.hpp

* Change vkb::core::HPPQueue from a facade over vkb::Queue to a self-contained class using vulkan.hpp

* Add transcoding of api sample hlsl_shaders based on vulkan.hpp

* Prepend "HPP" in NAME in CMakeLists.txt

plus modified hpp_hello_triangle/CMakeLists.txt accordingly to have a consistent naming scheme.

SaschaWillems committed:
Fix for graphics pipeline library sample (set required flags for fragment shader output pipeline library) (#508)

* Set required flags for fragment shader output pipeline library

Properly clean up fragment shaders created at runtime

* Removed trailing spaces

* Clangformat

marty-johnson59 committed:
Update build.md

Updating recommended Android CMake and NDK versions to avoid recent build issues

asuessenbach committed:
Re-enable vulkan_samples_pch; changing cmake_minimum_required to 3.16.

asuessenbach committed:
Merge branch 'precompile' of https://github.com/asuessenbach/Vulkan-Samples into precompile

asuessenbach committed:
Adjust copyright years.

asuessenbach commented:
It seems, moving to CMake version 3.16 worked!

asuessenbach committed:
Reuse the framework's precompiled headers in the hpp-samples.

gary-sweet reviewed (commented):
This change works ok for me, but it appears to also be adding a new sample (hpp_hlsl_shaders) which isn't mentioned anywhere in the description.

asuessenbach commented:
> but it appears to also be adding a new sample (hpp_hlsl_shaders) which isn't mentioned anywhere in the description.

That sample is already in... It's just the case that I rebased my changes after that PR was merged, and then had it in the list of commits... Don't know how to prevent such a noisance :(

SaschaWillems reviewed (approved):
This compiles fine for me now with VS2019. And build times seem to be at least a bit faster :)
